web/src/stream_edit.js

```diff
@@ -39,8 +39,20 @@ import * as user_groups from "./user_groups";
 import {user_settings} from "./user_settings";
 import * as util from "./util";
 
-function setup_subscriptions_stream_hash(sub) {
-    const hash = hash_util.stream_edit_url(sub);
+export const right_section_url_fragment_to_section_name = new Map([
+    ["personal", "personal_settings"],
+    ["subscribers", "subscriber_settings"],
+    ["general", "general_settings"],
+]);
+
+export const right_section_name_to_url_fragment = new Map([
+    ["personal_settings", "personal"],
+    ["subscriber_settings", "subscribers"],
+    ["general_settings", "general"],
+]);
+
+function setup_subscriptions_stream_hash(sub, right_section) {
```

Comment:
`right_section` can sometimes be not passed to this function, is it better to explicitly do `right_section=null` or something in those cases?

Reply:
(same question for some other functions in this call stack)

Reply:
Better might be to make `hash_util.stream_edit_url` expect a string always, and just pass `"general"` for this second parameter here?

Reply:
Yeah, I can explicitly pass "general" for all the other function calls